src/main/java/uk/gov/ons/ctp/response/action/service/CaseNotificationService.java

```diff
@@ -18,108 +17,134 @@
+        deleteActionCase(caseId);
+        break;
+
+      default:
+        log.with("notification_type", notification.getNotificationType())
+            .warn("Unknown case notification type", notification.getNotificationType());
+        break;
     }
-    final String sampleUnitIdStr = notification.getSampleUnitId();
-    final UUID sampleUnitId = sampleUnitIdStr == null ? null : UUID.fromString(sampleUnitIdStr);
-    final String sampleUnitType = notification.getSampleUnitType();
-
-    if (actionPlan != null) {
-      final ActionCase actionCase =
-          ActionCase.builder()
-              .id(caseId)
-              .sampleUnitId(sampleUnitId)
-              .actionPlanId(actionPlanId)
-              .actionPlanFK(actionPlan.getActionPlanPK())
-              .collectionExerciseId(collectionExerciseId)
-              .partyId(partyId)
-              .sampleUnitType(sampleUnitType)
-              .build();
-
-      switch (notification.getNotificationType()) {
-        case REPLACED:
-        case ACTIVATED:
-          final CollectionExerciseDTO collectionExercise =
-              collectionSvcClientServiceImpl.getCollectionExercise(collectionExerciseId);
-          actionCase.setActionPlanStartDate(
-              new Timestamp(collectionExercise.getScheduledStartDateTime().getTime()));
-          actionCase.setActionPlanEndDate(
-              new Timestamp(collectionExercise.getScheduledEndDateTime().getTime()));
-          checkAndSaveCase(actionCase);
-          break;
-
-        case DISABLED:
-        case DEACTIVATED:
-          try {
-            actionService.cancelActions(caseId);
-          } catch (final CTPException e) {
-            // TODO CTPA-1373 Do we really want to catch this. Should be let to go through.
-            // TODO CTPA-1373 What happens with other notif?
-            log.error(String.format("message = %s - cause = %s", e.getMessage(), e.getCause()));
-            log.error("Stacktrace: ", e);
-          }
-          final ActionCase actionCaseToDelete = actionCaseRepo.findById(caseId);
-          if (actionCaseToDelete != null) {
-            actionCaseRepo.delete(actionCaseToDelete);
-          } else {
-            log.warn("Unexpected situation where actionCaseToDelete is null for caseId {}", caseId);
-          }
-          break;
-        default:
-          log.warn("Unknown Case lifecycle event {}", notification.getNotificationType());
-          break;
-      }
-    } else {
-      log.warn("Cannot accept CaseNotification for non-existent actionplan {}", actionPlanIdStr);
+    actionCaseRepo.flush();
+  }
+
+  private ActionCase createActionCase(CaseNotification notification) throws CTPException {
+    UUID actionPlanId = UUID.fromString(notification.getActionPlanId());
+    ActionPlan actionPlan = actionPlanRepo.findById(actionPlanId);
+    if (actionPlan == null) {
+      throw new CTPException(CTPException.Fault.RESOURCE_NOT_FOUND, actionPlanId.toString());
```

Comment:
404 when trying to POST would be a bit weird. What's the logic here? Might be more like an `IllegalStateException`